tests/test_pipelines_audio_classification.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2021 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+import numpy as np
+
+from transformers import MODEL_FOR_AUDIO_CLASSIFICATION_MAPPING, PreTrainedTokenizer
+from transformers.pipelines import AudioClassificationPipeline, pipeline
+from transformers.testing_utils import is_pipeline_test, nested_simplify, require_datasets, require_tf, require_torch
+
+from .test_pipelines_common import ANY, PipelineTestCaseMeta
+
+
+@is_pipeline_test
+@require_torch
+class AudioClassificationPipelineTests(unittest.TestCase, metaclass=PipelineTestCaseMeta):
+    model_mapping = MODEL_FOR_AUDIO_CLASSIFICATION_MAPPING
+
+    @require_datasets
+    def run_pipeline_test(self, model, tokenizer, feature_extractor):
+        import datasets
+
+        audio_classifier = AudioClassificationPipeline(model=model, feature_extractor=feature_extractor)
+        dataset = datasets.load_dataset("anton-l/superb_dummy", "ks", split="test")
+
+        audio = np.array(dataset[0]["speech"], dtype=np.float32)
+        output = audio_classifier(audio)
+        self.assertEqual(
+            output,
+            [
+                {"score": ANY(float), "label": ANY(str)},
+                {"score": ANY(float), "label": ANY(str)},
+            ],
+        )
+
+    @require_torch
+    @require_datasets
+    def test_small_model_pt(self):
+        import datasets
+
+        model = "superb/wav2vec2-base-superb-ks"
+        tokenizer = PreTrainedTokenizer()
+        audio_classifier = pipeline("audio-classification", model=model, tokenizer=tokenizer)
```

Comment:
Any way to work around having an empty `tokenizer` here? Otherwise the `pipeline()` tries to load `Wav2Vec2CTCTokenizer` by default, since it's defined in the config mapper for `wav2vec2` and `hubert`.

Reply:
yeah for `"audio-classification"` we never need a tokenizer so we should change this here. The following:

```
 model = "superb/wav2vec2-base-superb-ks"
audio_classifier = pipeline("audio-classification", model=model)
```

doesn't work?

Reply:
Nope, an empty tokenizer falls into [this branch](https://github.com/huggingface/transformers/blob/master/src/transformers/pipelines/__init__.py#L447) which picks up `Wav2Vec2CTCTokenizer` linked in [tokenization_auto](https://github.com/huggingface/transformers/blob/c4ecd234f250cc6272812505b4a9b6cd1e29a816/src/transformers/models/auto/tokenization_auto.py#L171)

Reply:
Think it won't be that easy to **not** load a tokenizer for now. We could do an ugly: ``if task_class in ....`` -> don't load tokenizer here: https://github.com/huggingface/transformers/blob/42f359d015aee3835490bdcfa20df657a4d97049/src/transformers/pipelines/__init__.py#L444 but that's not the cleanest solution I think...maybe for now we can just silently load a tokenizer that won't be used. @Narsil what do you think?

Reply:
@anton-l - this should still be fine no? IMO it's not that problematic for now if a "useless" tokenizer is initialized (it's not pretty, but changing the general behavior here might require some more changes)

Reply:
Yeah, should be fine for demo purposes for now

Reply:
Agree with @Narsil